src/main/java/org/osiam/storage/query/UserFilterExpression.java

```diff
@@ -0,0 +1,81 @@
+package org.osiam.storage.query;
+
+import org.osiam.resources.scim.User;
+import org.osiam.storage.entities.UserEntity;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.util.Locale;
+
+public class UserFilterExpression extends FilterExpression<UserEntity> {
+
+    private Field field;
+
+    UserFilterExpression(String field, FilterConstraint constraint, String value) {
+        super(constraint, value);
+        this.field = new Field(field);
+    }
+
+    @Override
+    public UserFilterExpression.Field getField() {
+        return field;
+    }
+
+    public static class Field implements FilterExpression.Field<UserEntity> {
+        private static final Logger LOG = LoggerFactory.getLogger(UserFilterExpression.Field.class);
+        public static final String LOG_MESSAGE = "Period (.) used as field separator in %s. This is not SCIM conform and deprecated!";
+        private String urn;
+        private String fieldName;
+        private UserQueryField queryField;
+
+        public Field(String field) {
+            if (field.startsWith(User.SCHEMA)) {
```

Comment:
See the [comment above](https://github.com/osiam/osiam/pull/108/files#r50050541) for groups.